### DISCUSSION TITLE
SX1302 support

### DISCUSSION THREAD

**Discussion by PureTek-Innovations** [jgromes/RadioLib#887] [Ideas]
Hi
I use LoRa, not LoRaWAN for large-scale monitoring.
I use the Loramesher library.
I'd like to expand the things we monitor, but this will generate more packets, so I'd like to use more channels to reduce packet collisions.
The obvious upgrade is to multi-channel transceivers like the SX1302.
I don't think that I necessarily need to use RPIs, ESP32s are probably powerful enough, but I can use either.
Is it likely that RadioLib will ever support SX1302s or similar?
Thanks

**Comment by jgromes** [comment]
Unfortunately I don't think it's very likely. The primary reason is that there is no documentation of the SPI interface. The appropriate datasheet section is basically "just use our code lol":

![Screenshot_76](https://github.com/jgromes/RadioLib/assets/3407192/f5f32f60-817b-4bf1-933a-7b39f9ca11bb)

So the only way to implement SX1302 support would be to port the SX1302 HAL into RadioLib - I don't see much utility in that, honestly. Also, the SX1302 is not really comparable to the other transceivers supported by RadioLib, since it requires external RF frontends like SX1255 or SX1257.

**Reply by PureTek-Innovations** [reply]
Thanks
I was actually thinking of something like the RAK2287, I've got a couple (and their adapters) lying around.
https://docs.rakwireless.com/Product-Categories/WisLink/RAK2287/Datasheet/#overview
I have tried to get them to talk to an ESP32, I'll have another look at it this winter...

**Reply by jgromes** [reply]
All of these M.2 SX1302 gateways are based on the same reference design with the above mentioned RF fronteds. At the moment, I'm not aware of any other SX1302 driver apart from the [official one](https://github.com/Lora-net/sx1302_hal), which AFAIK only works on Linux.

**Reply by PureTek-Innovations** [reply]
I found this one
https://github.com/coversb/sx1302_esp32
I couldn't get it to work, so built a chirpstack gateway with the same adapter and concentrator to check the hardware was good.
Obviously with an RPI rather than an ESP32.
That worked, then it got shelved for a bit.
I'll get back to it at some point.

**Reply by berlincount** [reply]
> I found this one https://github.com/coversb/sx1302_esp32 I couldn't get it to work, so built a chirpstack gateway with the same adapter and concentrator to check the hardware was good. Obviously with an RPI rather than an ESP32. That worked, then it got shelved for a bit. I'll get back to it at some point.

I'm looking into getting the SX1302 in a MatchX M2 Pro repurposed as a Meshtastic gateway, so I'd be interested in collaborating - or at least learning from your findings, in case you want to share any further work ...

**Reply by Burritobun** [reply]
> > I found this one https://github.com/coversb/sx1302_esp32 I couldn't get it to work, so built a chirpstack gateway with the same adapter and concentrator to check the hardware was good. Obviously with an RPI rather than an ESP32. That worked, then it got shelved for a bit. I'll get back to it at some point.
> 
> I'm looking into getting the SX1302 in a MatchX M2 Pro repurposed as a Meshtastic gateway, so I'd be interested in collaborating - or at least learning from your findings, in case you want to share any further work ...

I'm doing the same thing. Have you began work on this? My plan was to not only get the sx1302 concentrator working, but to configure 125khz channel bonding to get the 250khz BW that Meshtastic requires. I'm not so much interested in using a light host as the stations have RPIs onboard.

**Reply by HeadBoffin** [reply]
To expand on our glorious leaders comment above, the support for the LoRa concentrator chip sets boils down to using the Semtech code base, so **IF** RadioLib were to support it, it would just be a wrapper on the Semtech code which would require a shed load of API calls that are unique to this chipset. So like strapping a jet engine to a petrol engine - potential to make things happen faster / in parallel - but the controls will be a mess.

The 'sensible' solution would be to fork any number of the Semtech SX1302 implementations and work on that. I did that to create a half-duplex version for LoRaWAN devices using ABP with a backhaul using RFC1149.
